applications/StructuralMechanicsApplication/custom_response_functions/adjoint_conditions/adjoint_semi_analytic_point_load_condition.cpp

```diff
@@ -0,0 +1,212 @@
+
+    Condition::Pointer AdjointSemiAnalyticPointLoadCondition::Create(Condition::Pointer pPrimalCondition ) const
+    {
+        return Kratos::make_shared<AdjointSemiAnalyticPointLoadCondition>( pPrimalCondition );
+    }
+
+    //******************************* DESTRUCTOR *****************************************
+    //************************************************************************************
+
+    AdjointSemiAnalyticPointLoadCondition::~AdjointSemiAnalyticPointLoadCondition()
+    {
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+
+    void AdjointSemiAnalyticPointLoadCondition::EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& rCurrentProcessInfo )
+    {
+        KRATOS_TRY
+
+        const unsigned int number_of_nodes = GetGeometry().size();
+        const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+        if (rResult.size() != dim * number_of_nodes)
+        {
+            rResult.resize(dim*number_of_nodes,false);
+        }
+
+        const unsigned int pos = this->GetGeometry()[0].GetDofPosition(ADJOINT_DISPLACEMENT_X);
+
+        if(dim == 2)
+        {
+            for (unsigned int i = 0; i < number_of_nodes; ++i)
+            {
+                const unsigned int index = i * 2;
+                rResult[index    ] = GetGeometry()[i].GetDof(ADJOINT_DISPLACEMENT_X,pos    ).EquationId();
+                rResult[index + 1] = GetGeometry()[i].GetDof(ADJOINT_DISPLACEMENT_Y,pos + 1).EquationId();
+            }
+        }
+        else
+        {
+            for (unsigned int i = 0; i < number_of_nodes; ++i)
+            {
+                const unsigned int index = i * 3;
+                rResult[index    ] = GetGeometry()[i].GetDof(ADJOINT_DISPLACEMENT_X,pos    ).EquationId();
+                rResult[index + 1] = GetGeometry()[i].GetDof(ADJOINT_DISPLACEMENT_Y,pos + 1).EquationId();
+                rResult[index + 2] = GetGeometry()[i].GetDof(ADJOINT_DISPLACEMENT_Z,pos + 2).EquationId();
+            }
+        }
+        KRATOS_CATCH("")
+    }
+
+    //***********************************************************************
+    //***********************************************************************
+
+    void AdjointSemiAnalyticPointLoadCondition::GetDofList(DofsVectorType& ElementalDofList, ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY
+
+        const unsigned int number_of_nodes = GetGeometry().size();
+        const unsigned int dim =  GetGeometry().WorkingSpaceDimension();
+        ElementalDofList.resize(0);
+        ElementalDofList.reserve(dim * number_of_nodes);
+
+        if(dim == 2)
+        {
+            for (unsigned int i = 0; i < number_of_nodes; ++i)
+            {
+                ElementalDofList.push_back( GetGeometry()[i].pGetDof(ADJOINT_DISPLACEMENT_X));
```

Comment:
Why not using the same as in the function above (giving the dof-position)?
this is faster

Reply:
Done